tests/test_sa_engine.py

```diff
@@ -147,3 +148,46 @@ async def test_terminate_with_acquired_connections(make_engine):
     await engine.wait_closed()
 
     assert conn.closed
+
+
+async def test_release_after_connection_disconnected_before_select(
+    tcp_proxy, unused_port, pg_params, make_engine
+):
+    server_port = pg_params["port"]
+    proxy_port = unused_port()
+
+    tcp_proxy = await tcp_proxy(proxy_port, server_port)
+    engine = await make_engine(port=proxy_port)
+
+    with pytest.raises(
+        (psycopg2.InterfaceError, psycopg2.OperationalError)
+    ):
+        with pytest.warns(ResourceWarning, match='Invalid transaction status'):
```

Comment:
Is it ok to spam users with both warning and exception? Maybe it is better just to raise an exception? Connection reset is not a user's fault, so I think that this warning doesn't help them.

Reply:
Thanks for noticing this. 

I'm a newbie in the aiopg codebase and don't know the reasons why these warnings are reported in such a way.

@asvetlov Could you share your opinion please?

Reply:
My initial thought was preventing a programming error and signal this situation early.
I agree that the connection releasing can be done for network reasons, there no need for extra checks.
I see two options:
1. Drop exception in 'release' as PR does, https://github.com/aio-libs/aiopg/blob/master/aiopg/pool.py#L242-L248 warning also can be dropped. It can lead to reports about unfinished transactions by **next** acquired connection.
Satisfactory but confusing.
2. Raise an exception only if the underlying socket is not closed but swallow the transaction status for disconnected peers. It prevents false positives from network errors but can report a programming error fast. Connection for the unclosed transaction should be dropped immediately, the connection is in an undefined incorrect state and cannot be recovered. Perhaps a warning generated in the pool can be replaced with a strict error also.

The 1) is easier, 2) is harder to implement but maybe more correct.
I don't want to decide what to do; I have no time to contribute anyway.  Please choose what you want.

Reply:
Okay, got you, thanks.

I've opened a new [issue](https://github.com/aio-libs/aiopg/issues/761) to fix it a bit later, because it looks like merging(testing) of the PR with incorrect warnings in quite rare case is much better than suffering from broken pool.

Reply:
Makes sense